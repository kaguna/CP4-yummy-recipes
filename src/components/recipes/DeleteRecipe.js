import React, { Component } from 'react';
import axiosInstance from '../../AxiosInstance'
class DeleteRecipe extends Component {
    constructor(props){
        super(props);

            this.state = { mess: "", error:""}
    }

    deleteRecipeHandler = (event) => {
        event.preventDefault();
        axiosInstance.delete("/category/"+this.props.recipe.category_id+"/recipe/"+this.props.recipe.id)
        .then(response => {
                this.props.parent.setState({mess: response.data.message, error: "",});
                this.props.recipeAfterDelete();
                
            })
            .catch(error => {
                if (error.response) {
                    this.setState({error: error.response.data.message, mess: ""});
                    console.log(error.response.data.message)
                } else if (error.request) {
                    this.setState({error:"Can't connect to the server.Please check your connection and try again.", mess: ""});
                }
            });
    };
    render() {
        return (
            
            <div className="modal fade" id={this.props.recipe_id} tabindex = "-1" role="dialog"
                 aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
                <div className="modal-dialog" role="document">
                    <div className="modal-content">
                        <div class="modal-header">
                            <h5 className="modal-title" id="exampleModalLongTitle">
                            Delete Recipe ({this.props.recipe.recipe_name})
                            <a href="#" className="pull-right" data-dismiss="modal">X</a>
                            </h5>
                        </div>
                        <div className="modal-body">
                            {this.state.error?
                                <div className="alert alert-danger">{this.state.error}</div>: ""}
                            <div className=" col-sm-12">
                                <h4>Are you sure you want to delete recipe <label className="control-label"> {this.props.recipe.recipe_name}?</label></h4>
                            </div>
                        </div>
                                <div className="modal-footer">
                                <button data-dismiss="modal" className="btn btn-default">
                                        <i class="glyphicon glyphicon-remove"></i> No, Cancel
                                    </button>
                                    <button data-dismiss="modal" type="button" onClick={this.deleteRecipeHandler} className="btn btn-danger">
                                        <i class="glyphicon glyphicon-ok"></i> Yes, Delete
                                    </button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
export default DeleteRecipe;
