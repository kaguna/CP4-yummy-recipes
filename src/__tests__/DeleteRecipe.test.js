import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import DeleteRecipe from '../components/recipes/DeleteRecipe';

describe('Edit recipes component', () => {
  const props = {
    recipe: { category_id: 1,
      recipe_description: 'fghbjnk',
      recipe_name: 'cool stuff',
      recipeId: 1 },
  };
  const wrapper = shallow(<DeleteRecipe {...props} />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
