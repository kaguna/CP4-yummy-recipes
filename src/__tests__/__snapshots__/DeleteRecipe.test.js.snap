// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit recipes component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalCenterTitle"
  className="modal fade"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Delete Recipe (
          cool stuff
          )
          <a
            className="pull-right"
            data-dismiss="modal"
            href=""
          >
            X
          </a>
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <div
          className=" col-sm-12"
        >
          <h4>
            Are you sure you want to delete recipe 
            <label
              className="control-label"
            >
               
              cool stuff
              ?
            </label>
          </h4>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-default"
          data-dismiss="modal"
        >
          <i
            className="glyphicon glyphicon-remove"
          />
           No, Cancel
        </button>
        <button
          className="btn btn-danger"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <i
            className="glyphicon glyphicon-ok"
          />
           Yes, Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
